7_react/react1/week3/todo-app/src/component/TodoContainer.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react';
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import 'bootstrap/dist/css/bootstrap.min.css'
+import {FaCalendarAlt} from 'react-icons/fa';
+
+//function for calender icon
+function CustomInput({value,onClick}){
+    return(
+        <div className='input-group'>
+            <input type='text' className='form-control' value={value} onClick={onClick} readOnly/>
+            <div className='input-group-append'>
+                <span className='input-group-text'>
+                    <FaCalendarAlt/>
+                </span>
+            </div>
+        </div>
+    )
+}
+
+const TodoContainer = ({ jsonTodos }) => {
+  const [newTodo, setNewTodo] = useState('');
+  const [todos, setTodos] = useState([]);
+  const [selectedDate, setDate] = useState(new Date());
+
+  useEffect(() => {
+    setTodos(jsonTodos);
+  }, [jsonTodos]);
+
+  const handleAddTodo = () => {
+    if (newTodo.trim() !== '' ) {      
+        const newTodoItem = {
+        id: todos.length + 1, 
+        title: newTodo,
+        deadline:selectedDate,
+        completed: false
+      };
+      setTodos([...todos, newTodoItem]);
+      setNewTodo('');
+      
+    }else {
+        alert('Please create new items & deadline for the todo.');
+      }
+  };
+
+  const handleEdit = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, editing: true } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleEditChange = (e, id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, title: e.target.value } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleUpdate = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, editing: false } : todo
+    );
+    setTodos(updatedTodos);
+  };
+
+  const handleDelete = (id) => {
+    const updatedTodos = todos.filter(todo => todo.id !== id);
+    setTodos(updatedTodos);
+  };
+
+   const handleCheckboxChange = (id) => {
+    const updatedTodos = todos.map(todo =>
+      todo.id === id ? { ...todo, completed: !todo.completed } : todo
+    );
+    setTodos(updatedTodos);
+  };
+  
+
+
+  return (
+    <div className="todo-container"style={{backgroundColor: "#2FBC8F",width:"80%"}} >
```

Comment:
why you are using somewhere `className` but somewhere else `style` prop? I think you can style the whole app just using CSS